iosApp/iosApp/ContentView.swift

```diff
@@ -157,108 +160,118 @@ struct ContentView: View {
-        let entry = sheetIdentityEntry.stackEntry
-        return GeometryReader { proxy in
-            sheetSwitch(entry: entry)
-                .onAppear {
-                    recordSheetHeight(proxy.size.height)
-                }
-                .onChange(of: proxy.size.height) { newValue in
-                    recordSheetHeight(newValue)
+                    .onChange(of: $nearbyVM.navigationStack.wrappedValue.lastSafe()) { newEntry in
+                        DispatchQueue.main.async {
+                            withAnimation {
+                                animatedLastEntry = newEntry
+                            }
+                        }
+                    }
+                    .onChange(of: $nearbyVM.navigationStack.wrappedValue.lastSafe().sheetItemIdentifiable()?
+                        .id) { _ in
+                            selectedDetent = .halfScreen
+                    }
+                    .onAppear {
+                        recordSheetHeight(proxy.size.height)
+                    }
+                    .onChange(of: proxy.size.height) { newValue in
+                        recordSheetHeight(newValue)
+                    }
                 }
-                // Adding id here prevents the next sheet from opening at the large detent.
-                // https://stackoverflow.com/a/77429540
-                .id(sheetIdentityEntry.id)
-                .presentationDetents([.small, .halfScreen, .almostFull], selection: $selectedDetent)
-                .interactiveDismissDisabled(visibleNearbySheet == .nearby)
-                .modifier(AllowsBackgroundInteraction())
-        }
+            })
     }
 
-    private func sheetSwitch(entry: SheetNavigationStackEntry) -> some View {
+    @ViewBuilder
+    var navSheetContents: some View {
         NavigationStack {
-            switch entry {
-            case .alertDetails:
-                EmptyView()
+            VStack {
+                switch animatedLastEntry {
+                case .alertDetails:
+                    EmptyView()
 
-            case let .stopDetails(stop, _):
-                StopDetailsPage(
-                    viewportProvider: viewportProvider,
-                    stop: stop, filter: $nearbyVM.navigationStack.lastStopDetailsFilter,
-                    nearbyVM: nearbyVM
-                ).onAppear {
-                    let filtered = nearbyVM.navigationStack.lastStopDetailsFilter != nil
-                    visibleNearbySheet = entry
-                    screenTracker.track(
-                        screen: filtered ? .stopDetailsFiltered : .stopDetailsUnfiltered
-                    )
-                }
+                case let .stopDetails(stop, filter):
+                    // Wrapping in a TabView helps the page to animate in as a single unit
+                    // Otherwise only the header animates
+                    TabView {
+                        StopDetailsPage(
```

Comment:
@EmmaSimon These issues should be resolved now!

I took the approach of each stop details page (outgoing & incoming) owning their own filter and departure state and updating the shared state when either of these values change _only if that stop details page is the incoming one_. This doesn't feel totally ideal to me but seems like the most reliable way to ensure there is no shared data between the two pages.

Reply:
Actually looking into an issue this introduced when tapping on a row from unfiltered stop details 👀 but I expect this general approach to still work

Reply:
Nice, the current approach is solid, this all looks good! I did notice a few transition animation quirks now that I want to mention, but this is definitely something we can release now and tweak later with design input.
* Transitions between different stop pages on the map have the slide up animation, while transitions between vehicles don't really seem to fade, just pop to the new vehicle.
* Updating the stop filter now has a fade transition (and even a lil slide up/down of the direction picker if the size changes). I don't hate this, but not sure if it was intentional. It's also a bit funky because of how the components are colored, the route header has its own colored background in addition to the colored background fill, which becomes visible during the fade.

Reply:
Thanks! I spoke with design about the lack of animation between trip details pages - I avoided implementing it on this pass since supporting slide would involve resolving [other race condition issues](https://mbta.slack.com/archives/C062QNAJZ2M/p1727791334641539?thread_ts=1727789603.100029&cid=C062QNAJZ2M), and it is a lower priority page. I'll definitely make a ticket for it! 

The stop filter animation was an unintentional side effect a side effect of breaking the relationship between stop details page and the last filter binding, but it is actually very straightforward to remove so I can do that now.

Reply:
I'm totally in favor of not doing anything to get trip-trip animations working before we get rid of the separate trip page entirely, but it's probably still good to have a ticket somewhere marking it as known. Thanks!